workspaces/arborist/lib/edge.js

```diff
@@ -112,7 +112,31 @@ class Edge {
     if (node.hasShrinkwrap || node.inShrinkwrap || node.inBundle) {
       return depValid(node, this.rawSpec, this.#accept, this.#from)
     }
-    return depValid(node, this.spec, this.#accept, this.#from)
+
+    // If there's no override we just use the spec.
+    if (!this.overrides) {
```

Comment:
To satisfy the concern https://github.com/npm/cli/pull/7025/files#r1829757867 the above could be
```suggestion
    if (!this.overrides?.keySpec) {
```

Reply:
and the first check of the function:
```js
if (node.name !== this.#name) {
  return false
}
``` 
could become:
```js
if (node.name !== this.#name || !this.#from) {
  return false
}
```

Reply:
Done